backstopper-core/src/test/java/com/nike/backstopper/handler/ApiExceptionHandlerBaseTest.java

```diff
@@ -474,6 +475,109 @@ public void doHandleApiException_should_not_allow_error_uid_from_passed_in_extra
+                .withStackTraceLoggingBehaviorAs(stackTraceloggingBehavior)
+                .build();
+
+        // when
+        boolean result = handler.shouldLogStackTrace(500, filteredClientErrors, originalException, apiException, reqMock);
+
+        // then
+        Assertions.assertThat(result).isEqualTo(true);
+    }
+
+    @Test
+    public void shouldLogStackTrace_should_not_log_all_errors_when_StackTraceLoggingBehavior_is_Default_for_an_ApiException() {
+        // given
+        ApiExceptionHandlerBase<TestDTO> handler = new TestApiExceptionHandler();
+
+        List<ApiError> filteredClientErrors = Arrays.<ApiError>asList(new ApiErrorBase("SOME_500_API_ERROR", 99043,
+                "some message", 500), new ApiErrorBase("CUSTOM_400_API_ERROR", 99042, "some message", 400));
+        Exception originalException = new Exception("something erroneous");
+
+        ApiException apiException = ApiException
+                .newBuilder()
+                .withApiErrors(testProjectApiErrors.getGenericServiceError())
+                .build();
+
+        // when
+        boolean result = handler.shouldLogStackTrace(500, filteredClientErrors, originalException, apiException, reqMock);
+
+        // then
+        Assertions.assertThat(result).isEqualTo(false);
+    }
+
+    @Test
+    public void shouldLogStackTrace_should_log_all_errors_when_coreException_is_not_an_ApiException_and_statusCode_is_500() {
+        // given
+        ApiExceptionHandlerBase<TestDTO> handler = new TestApiExceptionHandler();
+
+        List<ApiError> filteredClientErrors = Arrays.<ApiError>asList(new ApiErrorBase("SOME_500_API_ERROR", 99043,
+                "some message", 500));
+        Exception originalException = new Exception("something erroneous");
+
+        Exception coreException = new Exception("I am not an ApiException");
+
+        // when
+        boolean result = handler.shouldLogStackTrace(500, filteredClientErrors, originalException, coreException, reqMock);
+
+        // then
+        Assertions.assertThat(result).isEqualTo(true);
+    }
+
+
+    @Test
+    public void shouldLogStackTrace_should_not_log_all_errors_when_coreException_is_not_an_ApiException_and_statusCode_is_400() {
+        // given
+        ApiExceptionHandlerBase<TestDTO> handler = new TestApiExceptionHandler();
+
+        List<ApiError> filteredClientErrors = Arrays.<ApiError>asList(new ApiErrorBase("SOME_400_API_ERROR", 99044,
+                "some message", 400));
+        Exception originalException = new Exception("something erroneous");
+
+        Exception coreException = new Exception("I am not an ApiException");
+
+        // when
+        boolean result = handler.shouldLogStackTrace(400, filteredClientErrors, originalException, coreException, reqMock);
+
+        // then
+        Assertions.assertThat(result).isEqualTo(false);
+    }
+
+    @Test
+    public void shouldLogStackTrace_should_not_log_all_errors_when_StackTraceLoggingBehavior_is_FORCE_NO_STACK_TRACE() {
```

Comment:
You're using a 400, so it wouldn't log the stack trace anyway. This should be using a 500 along with `FORCE_NO_STACK_TRACE` to prove that the StackTraceLoggingBehavior is doing something other than the default behavior.